### PR TITLE
Druuid.gen_range -- convert datetime range to id range

### PR DESCRIPTION
**Pull request by drewkit**
When scoping down date ranges by Druuid ids, we've had to pad the range on either end by a second or so to ensure that we have the whole range covered. But padding introduces the additional hazard of including records in query results that are expressly outside the desired range. We currently account for this, but it would be nice to have it built into the Druuid library to provide accurate lower and upper bound ids.

### CODE REVIEW

**Comment by drewish** [comment]
What do you think about using a range rather than an array for the return type?

File: druuid.rb [hunk 1]
```diff
@@ -38,6 +38,20 @@ def time uuid, epoch_offset = epoch
...
 
+    # Provides lower and upper bound ids based on datetime range provided.
+    #
+    # @param [Time] time of lower bound
+    # @param [Time] time of upper bound
+    # @return [Array] ids of lower and upper bound range
+    # @example
+    #   Druuid.id_bounds Time.utc(2015,2,3,12,2,34), Time.utc(2015,2,3,17,56,12)
+    #   # => [11936695196844032000, 11936873186336964607]
+    def id_bounds lower_datetime, upper_datetime
```
Comment (drewish) [inline druuid.rb]: What do you think about making the second param optional and just having it default to the lower if it's not provided?

**Comment by whiteotter** [comment]
I can see range as offering more flexibility between building raw sql queries (Range#begin and Range#end) and simply handing off the whole range in an activerecord context.

Druuid#gen_range

File: druuid.rb (hunk 1 above)
Comment (whiteotter) [inline druuid.rb]: I overlooked the fact that we may at times simply desire an upper bound or lower bound id for a given datetime. Perhaps the id range method would simply comprise of two additional methods `Druuid.gen_lower_bound` and `Druuid.gen_upper_bound`.

**Pull request closed without merging**